fdbserver/workloads/TenantManagementWorkload.actor.cpp

```diff
@@ -1271,7 +1283,8 @@ struct TenantManagementWorkload : TestWorkload {
 		}
 
 		// Make sure the list terminated at the right spot
-		ASSERT(tenants.size() == limit || localItr == self->createdTenants.end() || localItr->first >= endTenant);
+		ASSERT(tenants.size() == limit || localItr == self->createdTenants.end() || localItr->first >= endTenant ||
+		       (tenantGroup.present() && localItr->first < endTenant && tenantMapItr == tenants.end()));
```

Comment:
I think the best way to structure this that also checks a few extra cases is to continue advancing `localItr` once `tenantMapItr` is exhausted if a tenant group was specified. Then you could check that there are no tenant group matches.

This addresses the case of the top block above, and you wouldn't need any change to this ASSERT. It also detects the case where there are tenants found locally in the group after the last tenant you processed from the request.